fs/sync/sync.go

```diff
@@ -876,6 +876,10 @@ func (s *syncCopyMove) Match(ctx context.Context, dst, src fs.DirEntry) (recurse
 			s.processError(err)
 		}
 	case fs.Directory:
+		// Only record matched (src & dst) empty dirs when performing move
+		if !s.DoMove {
```

Comment:
This has skipped the `errors.New("can't overwrite file with directory")` test below for sync & copy.

I think the test should probably be just around the src directory recording so after the `if ok`

Reply:
That makes more sense! I've updated my PR.